src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -38,114 +147,191 @@ def initialize(disk, ptemplate, slots)
+            CustomSizeInput.new(@slots),
+            CustomRegion.new(@slots)
+          ]
         end
 
         def init
-          Yast::UI.ChangeWidget(Id(:max_size), :Enabled, false)
-          Yast::UI.ChangeWidget(Id(:manual_size), :Enabled, false)
+          self.value = :max_size
+          # trigger disabling the other subwidgets
+          handle("ID" => value)
         end
 
         def store
-          start_block = Yast::UI.QueryWidget(Id(:start_block), :Value)
-          end_block = Yast::UI.QueryWidget(Id(:end_block), :Value)
-          len = end_block - start_block + 1
-          bsize = @slot.region.block_size # where does this come from?
-          region = Y2Storage::Region.create(start_block, len, bsize)
-          @ptemplate.region = region
-        end
-
-        def max_size_term
-          term(
-            :LeftRadioButton,
-            Id(:max_size),
-            Opt(:notify),
-            # radio button text, %1 is replaced by size
-            Builtins.sformat(
-              _("Maximum Size (%1)"),
-              @max_size.to_human_string
-            )
-          )
+          v = value
+          @ids ||= items.map(&:first)
+          w = widgets[@ids.index(v)]
+          w.store
+          @ptemplate.region = w.region
         end
+      end
 
-        def manual_size_term
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_size),
-            Opt(:notify),
-            _("Custom Size"),
-            VBox(
-              Id(:manual_size_attachment),
-              MinWidth(
-                15,
-                InputField(Id(:size_input), Opt(:shrinkable), _("Size"))
-              )
-            )
-          )
+      # An invisible widget that knows a Region
+      class MaxSizeDummy < CWM::Empty
+        attr_reader :region
+
+        def initialize(region)
+          @region = region
         end
 
-        def manual_region_term
-          min_block = @disk.region.start
-          # FIXME: libyui widget overflow :-(
-          max_block = min_block + @disk.region.length - 1
-          start_block = @slot.region.start
-          end_block = start_block + @slot.region.length - 1
-
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_region),
-            Opt(:notify),
-            _("Custom Region"),
-            VBox(
-              Id(:manual_region_attachment),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:start_block),
-                  _("Start Block"),
-                  min_block, max_block, start_block
-                )
-              ),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:end_block),
-                  _("End Block"),
-                  min_block, max_block, end_block
-                )
+        def store
+          # nothing to do, that's OK
+        end
```

Comment:
if you do not want to do anything, CWM allows to simple do not have it. I see that you call it from that controller, but do you test that it is not called twice?